lib/rmt/mirror/repomd.rb

```diff
@@ -65,9 +65,16 @@ def parse_packages_metadata(metadata_references)
                     primary: RepomdParser::PrimaryXmlParser }
 
     metadata_references.map do |file|
+      next if file.type == :deltainfo && !RMT::Config.mirror_drpm_files?
       next unless xml_parsers.key? file.type
 
       xml_parsers[file.type].new.parse_file(file.local_path)
     end.flatten.compact
   end
+
+  def need_to_download?(ref)
+    return false if !RMT::Config.mirror_drpm_files? && ref.local_path.match?(/\.drpm$/)
+
+    super(ref)
```

Comment:
I'm wondering, is this really needed since `next if file.type == :deltainfo && !RMT::Config.mirror_drpm_files?` should not even add the references of the list of potential packages to check, no?

Reply:
I was sincerely thinking about this too.

This part is an additional guard rail, *if it happens that files other than `type == :deltainfo` also contain `.drpm` packages*. I mean, what would happen if a `primary` file also contains `.drpm`?

I'm not a specialist in *repomd* repositories, and I wanted to play safe here, but, of course, we can review it if we feel assured enough that only `deltainfo` files references `.drpm`.

Reply:
mhm good point. Then let it stay as it is. I think `.drpm` definitions should not in `primary.xml` since it will list the packages available and drpm is only a diff of existing packages but you never know how strange certain repositories are setup.